frontend/integration-tests/tests/kubevirt/non-admin.scenario.ts

```diff
@@ -0,0 +1,174 @@
+    await browser.get(`${appHost}/k8s/cluster/${resource}`);
+    await crudView.isLoaded();
+    const exists = await crudView.rowForName(nonAdminNS).isPresent();
+    if (!exists) {
+      await crudView.createYAMLButton.click();
+      await browser.sleep(3000);
+      await browser.wait(until.presenceOf($('.modal-body__field')));
+      await $$('.modal-body__field').get(0).$('input').sendKeys(nonAdminNS);
+      await $$('.modal-body__field').get(1).$('input').sendKeys(`test-name=${nonAdminNS}`);
+      await $('.modal-content').$('#confirm-action').click();
+    }
+  });
+
+  it('Nonadmin user cannot use vm dialog in other namespace', async() => {
+    await browser.get(`${appHost}/k8s/ns/${testName}/virtualmachines`);
+    // crudView.isLoaded doesn't work here.
+    await browser.sleep(6000);
+    expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+
+    // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+
+    await browser.get(`${appHost}/k8s/ns/${testName}/vmtemplates`);
+    await browser.sleep(6000);
+    expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+
+    await browser.get(`${appHost}/overview/ns/${testName}/`);
+    await browser.sleep(6000);
+    expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+  });
+
+  it('Nonadmin user can use vm dialog in its own namespace', async() => {
+    await browser.get(`${appHost}/k8s/ns/${nonAdminNS}/virtualmachines`);
+    await crudView.isLoaded();
+    await click(crudView.createItemButton);
+    await click(createWithWizardLink);
+    await crudView.isLoaded();
+    await click(closeWizard);
+
+    await click(crudView.createItemButton);
+    await click(createWithYAMLLink);
+    await yamlView.isLoaded();
+
+    await browser.get(`${appHost}/k8s/ns/${nonAdminNS}/vmtemplates`);
+    await crudView.isLoaded();
+    await click(crudView.createItemButton);
+    await click(createWithWizardLink);
+    await crudView.isLoaded();
+    await click(closeWizard);
+
+    await browser.get(`${appHost}/overview/ns/${nonAdminNS}/`);
+    await crudView.isLoaded();
+  });
+
+  it('Nonadmin user with RoleBinding view can view vm objetcs in other NS', async() => {
+    execSync(`oc adm policy add-role-to-user view ${BRIDGE_HTPASSWD_USERNAME} -n ${testName}`);
+    await browser.get(`${appHost}/k8s/ns/${testName}/virtualmachines`);
+    await crudView.isLoaded(); // no error indicates it can view resource.
+    await browser.get(`${appHost}/k8s/ns/${testName}/vmtemplates`);
+    await crudView.isLoaded();
+    await browser.get(`${appHost}/overview/ns/${testName}/`);
+    await crudView.isLoaded();
+    execSync(`oc adm policy remove-role-from-user view ${BRIDGE_HTPASSWD_USERNAME} -n ${testName}`);
+  });
+
+  it('Nonadmin user with RoleBinding edit can create/edit vm objetcs in other NS', async() => {
+    execSync(`oc adm policy add-role-to-user edit ${BRIDGE_HTPASSWD_USERNAME} -n ${testName}`);
```

Comment:
this could be simplified by a function; e.g. called like this:

```ts
withRole(role, name, namespace, () => {
  await browser.get(`${appHost}/k8s/ns/${testName}/virtualmachines`);
  await crudView.isLoaded();
  ...
})
```

and similarly 
```ts
withClusterRole(...
```